modules/gateway/peers.go

```diff
@@ -173,20 +174,104 @@ func (g *Gateway) threadedAcceptConn(conn net.Conn) {
 	defer g.threads.Done()
 
 	addr := modules.NetAddress(conn.RemoteAddr().String())
-	g.log.Printf("INFO: %v wants to connect", addr)
+	g.log.Debugf("INFO: %v wants to connect", addr)
 
 	remoteVersion, err := acceptConnVersionHandshake(conn, build.Version)
 	if err != nil {
 		g.log.Debugf("INFO: %v wanted to connect but version handshake failed: %v", addr, err)
-		// TODO: should this be muxado.Server(conn).Close()?
 		conn.Close()
 		return
 	}
 
+	switch {
+	case build.VersionCmp(remoteVersion, "0.6.1") < 0:
+		err = g.managedAcceptConnOldPeer(conn, remoteVersion)
+	default:
+		err = g.managedAcceptConnNewPeer(conn, remoteVersion)
+	}
```

Comment:
`switch` is overkill, stick with `if else` for now. This is COMPAT code anyway and will eventually be removed.